src/platform/screen.c

```diff
@@ -1,22 +1,141 @@
 #include "platform/screen.h"
-
-#include "SDL.h"
+#include "platform/dpi.h"
 
 #include "game/settings.h"
 #include "graphics/graphics.h"
 #include "graphics/screen.h"
 
+#include "SDL.h"
+#include <math.h>
+#include <stdlib.h>
+
 static struct {
     SDL_Window *window;
     SDL_Renderer *renderer;
     SDL_Texture *texture;
-} SDL = {0, 0, 0};
+} SDL;
 
 static struct {
     int x;
     int y;
+    int w;
+    int h;
+    int moving;
 } window_pos;
 
+static struct {
+    const int WIDTH;
+    const int HEIGHT;
+} MINIMUM = { 640, 480 };
+
+/* Since resize events are called from a separate thread, we need to prevent unwanted resizes. */
+static int prevent_threaded_resize_event = 1;
+
+static void platform_screen_position_window(int width, int height)
+{
+    int title_height = platform_dpi_get_title_height();
+
+    /* No position information, center window */
+    if ((!window_pos.x && !window_pos.y) || !window_pos.w || !window_pos.h) {
+        int display = SDL_GetWindowDisplayIndex(SDL.window);
+        SDL_SetWindowPosition(SDL.window, SDL_WINDOWPOS_CENTERED_DISPLAY(display), SDL_WINDOWPOS_CENTERED_DISPLAY(display));
+        SDL_GetWindowPosition(SDL.window, &window_pos.x, &window_pos.y);
+    } else if (width != window_pos.w || height != window_pos.h) {
```

Comment:
For my own reference: this code must go to `platform_screen_center_window`, and we call that function from here instead. Otherwise when a user presses `F5` to center whe window, the window positions won't be updated.